packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/templates/defaultSearchBoxInput.html

```diff
@@ -37,6 +37,8 @@
               <template if:false={isQueryEmpty}>
                 <lightning-button-icon
                   onclick={clearInput}
```

Comment:
We shouldn't have to manually handle a ENTER key event, pressing ENTER or for the matter SPACE is the same as the `onclick` on this button.

You can see that the issue isn't reproduced on an older version of quantic for example here: https://connect.coveo.com/s/global-search/%40uri#q=index%20ms%20teams&tab=All

Pressing ENTER or SPACE after tabbing to the focus on the clear button clears the inputs, closes the suggestions and does NOT trigger a new query.

So I think the issue is with the refactor of the searchbox that we did recently. Something isn't right after that?

Reply:
The reason why we are manually handling the ENTER key event is that on the `clearInput` method, we are setting the focus back into the input in order to display the suggestions after clearing.

<img width="473" alt="image" src="https://github.com/coveo/ui-kit/assets/73316533/d4b8365b-b44d-4f6f-943e-82425fc9fef0">

However, this results in the `onClick` of the clear button to be triggered, calling `clearInput()` which will change focus on the `this.input`, will will trigger the keyup event on the input and thus calling `handleEnter()`, which will finally trigger the search after clearing. 

I believe that by manually handling the ENTER key event we can both 1) not trigger a search after clearing and 2) still getting to display the suggestions and keeping the focus on the input after clearing.

I tried a much simpler solution where I keep only the `onClick` listener on the clear button which clears the input, lose focus, does not trigger a search but does not display suggestions either...